scripts/train.py

```diff
@@ -1,93 +1,145 @@
-    regressor.compile(optimizer='rmsprop',loss='mean_squared_error'); print(regressor.summary())
-    history=regressor.fit(X,y,epochs=epoch, batch_size=batch_size)
+    # =======================================================================================================================
+    regressor.compile(optimizer='rmsprop', loss='mean_squared_error')
+    print(regressor.summary())
+    return regressor
+
+
+def train(input, output, regressor, batch_size=50, epochs=30):
+    """Train the model and display the training preprocess."""
+    history = regressor.fit(input, output, batch_size, epochs)
     print(history)
     return regressor
 
 
+def save_model(regressor, ticker='', epochs='UNKNOWN'):
+    """Save the trained models, with default unkown epochs."""
+    regressor.save('model_{}_{}_epochs.h5'.format(ticker, epochs))
 
-# it's better practice to define all functions before if __name__=='__main__'
-if __name__ == '__main__':
+
+def cli_args():
+    """Read arguments from CLI."""
     parser = argparse.ArgumentParser(description="Do something.")
-    #optional arguments are ID-ed by the - prefix, and the remaining arguments are assumed to be positional
-    parser.add_argument("--path", "-filepath"); parser.add_argument("--epochs", "-epoch")
+    # optional arguments are ID-ed by the - prefix, and the remaining arguments
+    # are assumed to be positional
+    parser.add_argument("--path", "-path")
+    parser.add_argument("--epochs", "-epoch")
+    parser.add_argument("--ticker", "-ticker")
     args = parser.parse_args(sys.argv[1:])
-    dataset=read_data(args.path)
+    return args
+
+
+if __name__ == '__main__':
+    args = cli_args()
+    dataset = read_data(args.path)
     print('Tail of the dataset is: \n\n {}:'.format(dataset.tail()))
+    training_set, test_set = train_test_split(dataset)
+    train_test_split_plot(dataset)
+
+    train_input, train_output = input_output_split(training_set)
 
-    training_set, test_set=train_test_split(dataset)
-    dataset["High"][:'2016'].plot(figsize=(16,4),legend=True)
-    dataset["High"]['2017':].plot(figsize=(16,4),legend=True)
-    plt.legend(['Training set (Before 2017)','Test set (2017 and beyond)'])
-    plt.title('IBM stock price');plt.show()
-
-    training_set_scaled = sc.fit_transform(training_set)
-    X_train, y_train=X_y_split(60) #X_train.shape=(2709, 60); y_train.shape=(2709, 1)
-    X_train, y_train=X_y_reshape(1,X_train, y_train)
-    X_train = np.reshape(X_train, (X_train.shape[0],X_train.shape[1],1)) # X_train.shape=(2709 , 60 , 1)
-    # 1st, 2nd, 3rd dimensions: the number of samples. time-steps fed to a sequence, features('price').
-    regressor= build(X_train, y_train, batch_size=50, epoch=int(args.epochs))
-    regressor.save('model_epoch_{}.h5'.format(args.epochs))
+    regressor = build(train_input)
+    regressor = train(train_input, train_output, regressor, epochs=int(args.epochs), batch_size=50)
```

Comment:
you can make `args.epochs` an `int` by default by specifying a type with `add_argument ` https://docs.python.org/3/library/argparse.html#type